templates/common/on-prem/units/api-int-service.yaml

```
@@ -0,0 +1,11 @@
+name: api-int.service
+# This service is run by a timer
+enabled: false
+contents: |
+  [Unit]
+  Description=Ensure api-int entry exists in /etc/hosts
+  Before=kubelet.service crio.service
+
+  [Service]
+  Type=oneshot
+  ExecStart=/bin/bash -c "grep -q {{ onPremPlatformAPIServerInternalIP . }} api-int api-int.{{ .DNS.Spec.BaseDomain }} /etc/hosts || echo '{{ onPremPlatformAPIServerInternalIP . }} api-int api-int.{{ .DNS.Spec.BaseDomain }}' >> /etc/hosts"
```

The search pattern should be quoted so that `api-int` and  `api-int.{{ .DNS.Spec.BaseDomain }}` are parsed as part of the pattern and not as names of files to search:
```suggestion
  ExecStart=/bin/bash -c "grep -q -e '{{ onPremPlatformAPIServerInternalIP . }} api-int api-int.{{ .DNS.Spec.BaseDomain }}' /etc/hosts || echo '{{ onPremPlatformAPIServerInternalIP . }} api-int api-int.{{ .DNS.Spec.BaseDomain }}' >> /etc/hosts"
```
Is there any chance the internal IP address or base domain could change?  If so, presumably you need to clean up the old entries.

---

Also, you might as well use `-F`:
```suggestion
  ExecStart=/bin/bash -c "grep -q -e '{{ onPremPlatformAPIServerInternalIP . }} api-int api-int.{{ .DNS.Spec.BaseDomain }}' -F /etc/hosts || echo '{{ onPremPlatformAPIServerInternalIP . }} api-int api-int.{{ .DNS.Spec.BaseDomain }}' >> /etc/hosts"
```

---

Good suggestions, thanks!

I'm not sure whether we support changing the api vip or the domain, but I think we can make this easy to clean up, so I'll make that change as well.